### PR TITLE
CXX-875 database::has_collection throws if operation fails

### PR DESCRIPTION
**Pull request by samantharitter**
@xdg PTAL?

Also it looks like some things in `helpers.hpp` fell victim to clang-format...

### CODE REVIEW

**Comment by Machyne** [comment]
LGTM

File: src/third_party/catch/include/helpers.hpp [hunk 1]
```diff
@@ -139,9 +137,9 @@
     find_and_modify_opts_set_bypass_document_validation->interpose(                                \
         [&expected_find_and_modify_opts_bypass_document_validation](                               \
             mongoc_find_and_modify_opts_t*, bool bypass_document_validation) {                     \
-            REQUIRE(bypass_document_validation ==                                                  \
-                    expected_find_and_modify_opts_bypass_document_validation);                     \
-            return true;                                                                           \
+        REQUIRE(bypass_document_validation ==                                                      \
```
Comment (hanumantmk) [inline src/third_party/catch/include/helpers.hpp]: What happened with all of the indentation?

Comment (hanumantmk) [inline src/third_party/catch/include/helpers.hpp]: ah, I saw your comment, nvm

**Comment by hanumantmk** [comment]
LGTM

**Pull request merged**